scripts/gen_vimdoc.py

```diff
@@ -563,21 +564,23 @@ def parse_parblock(parent, prefix='', width=62, indent=''):
 # }}}
 
 
-def parse_source_xml_for_mpack(filename, mode):
-    """Collects API functions to be packed as msgpack.
+def extract_from_xml(filename, mode, fmt_vimhelp):
+    """Extracts Doxygen info as maps without formatting the text.
 
     Returns two maps:
-      1. API functions
-      2. Deprecated API functions
+      1. Functions
+      2. Deprecated functions
 
-    Caller decides what to do with the deprecated documentation.
+    The `fmt_vimhelp` parameter controls some special cases for use by
+    fmt_doxygen_xml_as_vimhelp(). (TODO: ugly :)
```

Comment:
What do you think of building the Vim help text in `fmt_doxygen_xml_as_vimhelp()` directly from the function maps returned by `extract_from_xml`? That way there’s no need for the fmt_vimhelp and desc_node stuff. Only thing is that I’m unsure how much more work that’d be.

Reply:
That would be ideal.